UI/js/users/users.js

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (C) 2018
+Developed at University of Toronto
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+var userRow = null;
+var userList = null;
+
+$(function () {
+    $('select').material_select();
+
+    $('#typeFilter').on('change', function () {
+        startLoad('#usersLoad', '#usersList');
+        displayList();
+    });
+
+    $('#statusFilter').on('change', function () {
+        startLoad('#usersLoad', '#usersList');
+        displayList();
+    });
+
+    $('#searchFilter').on('keyup', function () {
+        startLoad('#usersLoad', '#usersList');
+        displayList();
+    });
+
+    getUsersList();
+    startLoad('#usersLoad', '#usersList');
+    getUsersList();
+});
+
+function getUsersList() {
+    $.ajax({
+        type: 'GET',
+        url: '/usersListComponent',
+        success: function (data) {
+            userRow = $(data.usersEntryHTML);
+            userList = data.usersList;
+            displayList();
+        },
+        error: function (data) {
+            //TODO: add fail snackbar
+        }
+    });
+}
+
+function displayList() {
+    $('#usersList').html('');
+    var rowPopulate = '';
+
+    userList.forEach(user => {
+        if (passFilter(user)) {
+            $('#usersList').append(fillRow(user));
+        }
+    });
+
+    if ($('#usersList').find('li').length === 0) {
+        $('#usersList').append(`<p class="center"><i>${translate('noResultsFoundBasedOnSearch')}</i></p>`)
+    }
+
+    endLoad('#usersLoad', '#usersList');
+}
+
+function fillRow(user) {
+    var bindedRow = userRow;
+    var color = colours.green;
+    var status = user.status;
+
+    if (status === 0) {
```

Comment:
For this and below, use userStatus const for comparison

Reply:
Not sure what you mean

Reply:
its that simple. the common.userStatus is a backend only object that cant be accessed directly from the UI. I am working on implementing a settings object that should solve this issue and others